app/src/main/java/com/mstudio/superheroarch/MainActivity.kt

```diff
@@ -1,36 +1,24 @@
 package com.mstudio.superheroarch
 
 import android.os.Bundle
-import android.view.LayoutInflater
 import android.view.View
-import android.view.ViewGroup
 import android.widget.Button
-import android.widget.ImageView
-import android.widget.TextView
 import androidx.activity.enableEdgeToEdge
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
-import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
 import com.google.android.material.snackbar.Snackbar
+import com.mstudio.superheroarch.ApiService.retrofit
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
-import com.squareup.picasso.Picasso
-import retrofit2.Response
-import retrofit2.Retrofit
-import retrofit2.converter.gson.GsonConverterFactory
-import retrofit2.http.GET
-
 class MainActivity : AppCompatActivity() {
 
-    val retrofit = Retrofit.Builder()
-        .baseUrl("https://rickandmortyapi.com/api/")
-        .addConverterFactory(GsonConverterFactory.create())
-        .build()
-
+    private lateinit var characterRecyclerView: RecyclerView
```

Comment:
do you need the lateinit vars?

Reply:
It is used here because RecyclerView cannot be initialized until setContentView has been called and the inflated view is available, but it can also be done without using lateinit.

Reply:
Changed, there is no longer lateinit variable